Hotfix/pqpf

DingchenHou-NOAA opened the pull request.

XianwuXue-NOAA committed:
Add back global_enspqpf.fd and build_global_enspqpf.sh

 On branch hotfix/pqpf
	new file:   build_global_enspqpf.sh
	new file:   global_enspqpf.fd/Makefile
	new file:   global_enspqpf.fd/PQPF.f90
	new file:   global_enspqpf.fd/init_parm.f90
	new file:   global_enspqpf.fd/printinfr.f90
Refs: #2

DingchenHou-NOAA committed:
Merge branch 'hotfix/pqpf' of github.com:XianwuXue-NOAA/GEFS into hotfix/pqpf

XianwuXue-NOAA committed:
Add module files for enspqpf program

 On branch hotfix/pqpf
	new file:   modulefiles/gefs/global_enspqpf.hera
	new file:   modulefiles/gefs/global_enspqpf.jet
	new file:   modulefiles/gefs/global_enspqpf.wcoss_cray
	new file:   modulefiles/gefs/global_enspqpf.wcoss_dell_p3

Refs: #2

XianwuXue-NOAA committed:
Add global_enspqpf to build workflow in sore
 On branch hotfix/pqpf
	modified:   build_all.sh
	modified:   gefs_build.cfg
	modified:   partial_build.sh

Refs: #2

DingchenHou-NOAA committed:
Merge branch 'hotfix/pqpf' of github.com:XianwuXue-NOAA/GEFS into hotfix/pqpf

DingchenHou-NOAA committed:
Add pqpf procudure back to the enspost job

In GEFS v12 implementation, the application of pgpf procudure was accidently
removed from the ex-script of the atmos_enspost job and consquently, the
follwing files, their grb2 version as well as the corresponding index
files (20 files in total) are missing from the the atmos/ensstat directory
gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqff
gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqif
gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqpf
gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqrf
gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqsf

Scripts of the atmos_enspost jobs are modified to fix this problem and
other minor issues:
In the J-job script, $COMIN is distinguished from $COMOUT
In the ex-script, the pqpf block is added back for hr job (0-16d, ext_h= )
In the ush script, nepert is added to the input parameter list

Refs: #2

WalterKolczynski-NOAA commented on scripts/exgefs_atmos_enspost.sh:
```diff
@@ -460,7 +468,24 @@ if [[ "$cycle" == "t00z" ]] && [[ -z $ext_h ]]; then
 # part (1b): probabilistic forecasts ( PQPF, PQRF, PQFF, PQSF and PQIF )
+if [[ -z $ext_h ]]; then
+
 export CDATE=$PDY$cyc; 
```
This block is not properly indented

WalterKolczynski-NOAA commented on scripts/exgefs_atmos_enspost.sh:
```diff
@@ -460,7 +468,24 @@ if [[ "$cycle" == "t00z" ]] && [[ -z $ext_h ]]; then
+$ENSPQPF
+
+for file in pqpf pqrf pqff pqsf pqif; do
+    $CNVGRIB -g21 $DATA/$file $DATA/$file\_grb1
```
Use { } to denote the variable name boundary rather than escaping the underscore, as is done other places in this block.

XianwuXue-NOAA replied:
agree

WalterKolczynski-NOAA commented on sorc/global_enspqpf.fd/PQPF.f90:
```diff
@@ -0,0 +1,399 @@
+call getgb2(11,0,jskp,jdisc,jids,jpdtn,jpdt,jgdtn,jgdt,unpack,jskp,gfldo,iret)
+
+if(iret.ne.0) then; print*,' getgbeh, cannot get maxgrd ';endif
+if(iret.ne.0) goto 1020
```
This will need to be rewritten to remove the goto statements.

XianwuXue-NOAA replied:
I modified this file to remove the "goto"statement, however, I used wrong reference number (#3 for this PR instead of #2), so Please check my changes and if it is OK, Dingchen can merge it into his branch for this PR.

WalterKolczynski-NOAA reviewed (changes requested):
Needs some minor changes, mostly for formatting.

XianwuXue-NOAA committed:
Remote goto statement in global_enspqpf program

 On branch hotfix/pqpf
 Changes to be committed:
	modified:   sorc/global_enspqpf.fd/PQPF.f90

Refs: #3

XianwuXue-NOAA committed:
Merge remote-tracking branch 'dingchen/hotfix/pqpf' into hotfix/pqpf

XianwuXue-NOAA committed:
Delete unused statements

 On branch hotfix/pqpf
	modified:   sorc/build_global_enspqpf.sh

Refs: #2

DingchenHou-NOAA committed:
Minor changes to improve pqpf procedure

In the scripts/exgefs_atmos_enspost.sh script, improved indentation
and variable name boundary

in the pqpf code file, updated information about ensemble size

Refs: #2

XianwuXue-NOAA committed:
Add all changes for global_enspqpf

1) 6c32ae9 on Nov 25, 2020 by Dingchen

 In GEFS v12 implementation, the application of pgpf procudure was accidently
 removed from the ex-script of the atmos_enspost job and consquently, the
 follwing files, their grb2 version as well as the corresponding index
 files (20 files in total) are missing from the the atmos/ensstat directory
 gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqff
 gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqif
 gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqpf
 gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqrf
 gefs.${PDY}/${cyc}/atmos/ensstat/ensstat.t00z.pqsf

 Scripts of the atmos_enspost jobs are modified to fix this problem and
 other minor issues:
 In the J-job script, $COMIN is distinguished from $COMOUT
 In the ex-script, the pqpf block is added back for hr job (0-16d, ext_h= )
 In the ush script, nepert is added to the input parameter list

2) c8f2ca9 on Nov 30, 2020 by Xianwu Xue

 Delete unused statements in sorc/build_global_enspqpf.sh

3) 1451baf on Nov 30, 2020

 In the scripts/exgefs_atmos_enspost.sh script, improved indentation
 and variable name boundary
 in the pqpf code file, updated information about ensemble size

 On branch hotfix/pqpf_ops
	modified:   jobs/JGEFS_ATMOS_ENSPOST
	modified:   scripts/exgefs_atmos_enspost.sh
	modified:   sorc/build_global_enspqpf.sh
	modified:   ush/global_enspqpf.sh

Refs: #2

DingchenHou-NOAA committed:
Merge remote-tracking branch 'XX/hotfix/pqpf_ops' into hotfix/pqpf

DingchenHou-NOAA commented:
Working with Xianwu, I have addressed the comments.
